contracts/dnsregistrar/DNSRegistrar.sol

```diff
@@ -98,11 +85,39 @@ contract DNSRegistrar {
         claim(name, proof);
     }
 
+    function proveAndClaimWithResolver(bytes memory name, DNSSEC.RRSetWithSignature[] memory input, bytes memory proof, address resolver, address addr) public {
+        proof = oracle.submitRRSets(input, proof);
+        (bytes32 rootNode, bytes32 labelHash, address owner) = _claim(name, proof);
+        require(msg.sender == owner, "Only owner can call proveAndClaimWithResolver");
```

Comment:
We no longer allow anyone else to claim? If this is due to the risk that someone could set a custom resolver to do something malicious, we could also create a function like `proveAndClaimWithDefaultResolver` which takes the default resolver by looking up `resolver.ens.eth`.

Reply:
We don't allow it because there's no way to know what the owner wanted unless it's them calling it. If we allow setting the default resolver, anyone could then override a custom resolver by calling this.

Reply:
I update our keybase conversation. We don't do this as there are edge cases like `anyone who does have a resolver set could then have it replaced with the standard resolver by anyone else` and doing it for only names which haven't set resolver cause another issue of `what if the owner changes? Can you do it? Can only the owner?`